### ISSUE TITLE
Firefox window with titlebar has black background box offset region (similar to an ugly shadow)

### ISSUE THREAD

**Issue by worldofpeace** [elementary/gala#948]
## What Happened

This has been reported at stylesheet https://github.com/elementary/stylesheet/issues/845 but I have reason to believe it might not be stylesheet related because other CSD gtk apps don't have the problem. It's just firefox. 

![Screenshot from 2020-10-27 03 31 04](https://user-images.githubusercontent.com/28888242/97270069-e40e2480-1804-11eb-85de-ed026682a6cf.png)

The catch is it will go away when fullscreen or **only** if you enable the broken csd for Firefox, where the close and maximize buttons are both on the left instead of split on each side. It seems at the time of implementing csd they only tested button orientations for unity :rofl: In chromium it works perfectly, though.

Another thing is in what environment I noticed this. I upgraded gala to master and built with mutter 3.36, and only then did it trigger the issue. I tried to look at Mutter GitLab if there's a reported fix but I didn't find one. If this can't be reproduce on focal then maybe there is a patch and Ubuntu has it for Mutter :shrug: 

Lastly, this is reported in the Firefox tracker, it could be a Firefox bug for all I know https://bugzilla.mozilla.org/show_bug.cgi?id=1666839. But considering they marked it as `wontfix` I decided it would be best to get some engagement going here.

## Expected Behavior

Not have this ugly offset region

## Steps to Reproduce

1. Build gala on master with Mutter 3.36
2. Open Firefox floating without csd 
3. See weird black background box offset region

### Logs

NA

## Platform Information

NixOS 21.03 pre-git
Compiled from git, whatever commit is dated 2020-10-23.
GTK 3.24.21
GNOME 3.36 Stack

**Comment by finetuned89** [comment]
I have this same issue on FireFox, but in my case, only when running a web site in "App Mode" (SSB). It started happening a few weeks ago for me.

**Comment by fidelisakilan** [comment]
i have a similar issue if i remove titlebar its gone then downloads has a separate window which has black square behind it

**Comment by tintou** [comment]
It would be great to be able to know if:
 * This is a bug from Gala: can someone replicate the issue in GNOME Shell with the latest elementary Stylesheet, is there the same issue with Adwaita using Gala
 * This is a bug in the new Stylesheet only

Reporting the tests on Firefox side too might be a good thing

**Comment by Dirli** [comment]
i have gentoo, debian:
stylesheet (5.4.2) and gala (3.3.2)
gentoo: firefox 82 - there is this bug
debian: firefox 78 (esr) - this bug is missing
I think this is a firefox problem

**Comment by worldofpeace** [comment]
Yeah, that proves as much that this is a firefox bug. I would still keep this open here, though. 

@tintou I think the things @Dirli checked, being that firefox ESR is without the issue, is evidence enough.

**Comment by worldofpeace** [comment]
> It would be great to be able to know if:
> 
>     * This is a bug from Gala: can someone replicate the issue in GNOME Shell with the latest elementary Stylesheet, is there the same issue with Adwaita using Gala
> 
>     * This is a bug in the new Stylesheet only
> 
> 
> Reporting the tests on Firefox side too might be a good thing

I could do these checks. Though, Firefox ESR not having the problem might be our answer that Upstream just turned this away without enough information.

**Comment by finetuned89** [comment]
I checked the elementary stylesheet (from master branch) on Ubuntu 20.04, and I don't have the same problem there. How do I install the adwaita stylesheet on elementary?

**Comment by cassidyjames** [comment]
@finetuned89 it's installed by default; run:

```
gsettings set org.gnome.desktop.interface gtk-theme Adwaita
``` 

to enable it, and:

```
gsettings reset org.gnome.desktop.interface gtk-theme
```

to restore the default setting,.

**Comment by gamerlv** [comment]
I first noticed this bug after upgrading firefox to 81, firefox 80 does not have this issue. This issue only happens on windows with titlebars with me. CSD enabled windows look normal even on firefox 82. (Menu -> Customize, At the bottom uncheck "title bar")

The issue worldofpeace linked has a [commit attached](https://hg.mozilla.org/integration/autoland/rev/16bf991df0003caa9a5db83448373fb4f0bcd561) as possible regression. I've rebuild firefox 82 (firefox_82.0+build2-0ubuntu0.18.04.1) reversing this patch. This issue no longer shows on my system now.

**Comment by cassidyjames** [comment]
I see this issue in Firefox 82 (via Flatpak) on elementary OS 5.1 Hera as well—even with Adwaita. So my guess is it's some combination of a Firefox and Gala issue. :smile:

**Comment by worldofpeace** [comment]
Please also confirm it's using Mutter 3.36, is that available on ubuntu 20.04?

**Comment by Zonnev** [comment]
> Please also confirm it's using Mutter 3.36, is that available on ubuntu 20.04?

Ubuntu 20.04.1 GNOME runs on Mutter 3.36.6. Firefox 82.0.2 doesn't have this problem there.

**Comment by Waester** [comment]
Starting Firefox with GTK_CSD=0 fixed this for me. Workaround was found by Zbergen-cli at https://github.com/Zonnev/elementaryos-firefox-theme/issues/90#issuecomment-729619811

**Comment by worldofpeace** [comment]
> The catch is it will go away when fullscreen or only if you enable the broken csd for Firefox

This is mentioned in the OP

**Comment by Alecaddd** [comment]
Heads up that in Firefox 85+ this seems to be solved.
Anyone else available to confirm?

**Comment by CosCal** [comment]
Glad I found this. Been like this for months now.
elementary OS 5.1.7 Hera - Linux 5.4.0-56-generic (GTK 3.22.30)
AMD Ryzen7 3700X - Radeon 5700 - B550 Velocita
Shows up with stock AppCenter install. Firefox version 83.0 (64-bit)
![Shadow_Issue](https://user-images.githubusercontent.com/34515453/102119787-a9da0000-3e0f-11eb-8b5b-85c8447b7e1a.gif)

**Comment by shoccho** [comment]
> Heads up that in Firefox 85+ this seems to be solved.
> Anyone else available to confirm?
@Alecaddd 


the issue is not present in Firefox 
Version = 85.0
BuildID = 20210107185759


![Screenshot from 2021-01-09 16-42-45](https://user-images.githubusercontent.com/33720947/104089858-6488e100-529c-11eb-9b26-a550262a7d93.png)
left : v84
right : v85

**Comment by worldofpeace** [comment]
I will try this later today in NixOS

**Comment by lenemter** [comment]
Closing as stale. If you still can reproduce the issue in OS 8, please reply here and I'll reopen it
